test/ERC20RootVault.test.ts

```diff
@@ -0,0 +1,1149 @@
+                        this.deployer.address,
+                        this.deployer.address
+                    );
+                    const erc20Vault = await read(
+                        "VaultRegistry",
+                        "vaultForNft",
+                        erc20VaultNft
+                    );
+                    const uniV3Vault = await read(
+                        "VaultRegistry",
+                        "vaultForNft",
+                        uniV3VaultNft
+                    );
+                    const erc20RootVault = await read(
+                        "VaultRegistry",
+                        "vaultForNft",
+                        erc20VaultNft + 1
+                    );
+
+                    this.erc20Vault = await ethers.getContractAt(
+                        "ERC20Vault",
+                        erc20Vault
+                    );
+
+                    this.uniV3Vault = await ethers.getContractAt(
+                        "UniV3Vault",
+                        uniV3Vault
+                    );
+
+                    this.subject = await ethers.getContractAt(
+                        "ERC20RootVault",
+                        erc20RootVault
+                    );
+
+                    for (let address of [
+                        this.deployer.address,
+                        this.uniV3Vault.address,
+                        this.erc20Vault.address,
+                    ]) {
+                        await mint(
+                            "USDC",
+                            address,
+                            BigNumber.from(10).pow(18).mul(3000)
+                        );
+                        await mint(
+                            "WETH",
+                            address,
+                            BigNumber.from(10).pow(18).mul(3000)
+                        );
+                        await this.weth.approve(
+                            address,
+                            ethers.constants.MaxUint256
+                        );
+                        await this.usdc.approve(
+                            address,
+                            ethers.constants.MaxUint256
+                        );
+                    }
+
+                    await this.weth.approve(
+                        this.subject.address,
+                        ethers.constants.MaxUint256
+                    );
+                    await this.usdc.approve(
+                        this.subject.address,
+                        ethers.constants.MaxUint256
+                    );
+
+                    return this.subject;
+                }
+            );
+        });
+
+        beforeEach(async () => {
+            await this.deploymentFixture();
+        });
+
+        describe("#depositorsAllowlist", () => {
+            it("returns non zero length of depositorsAllowlist", async () => {
+                expect(
+                    (await this.subject.depositorsAllowlist()).length
```

Comment:
a little bit strange test. It is way better to set a depositorsAllowList and compare actual result with the expected one